File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque
def solution(progresses, speeds):
    answer = []
    p_queue = deque(progresses)
    s_queue = deque(speeds)
    
    while len(p_queue) > 0: #큐가 비어있지 않으면 
        
        #매일 작업 진도율 반영
        for i in range(len(p_queue)):
            p_queue[i] += s_queue[i]
            
        cnt = 0
        while p_queue and p_queue[0] >= 100: #순위가 높은 작업이 끝나면
            print(p_queue)
            p_queue.popleft()
            s_queue.popleft()
            cnt += 1
        if cnt > 0: answer.append(cnt)
        
        
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;
import java.math.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        
        // 완료일 계산
        int[] complete = new int[progresses.length];
        for(int i = 0; i<progresses.length; i++){
            complete[i] = (int)Math.ceil((double)(100 - progresses[i]) / speeds[i]);
        }
        
        List<Integer> list = new ArrayList<>();
        int prev = complete[0];
        int count = 1;
        for(int j = 1; j<complete.length; j++){
            if(prev < complete[j]){
                prev = complete[j];
                list.add(count);
                count = 1;
            }else{
                count++;
            }
        }
        list.add(count); //마지막 배포기능
        
        int[] answer = new int[list.size()];
        for(int d=0; d<answer.length; d++){
            answer[d] = list.get(d);
        }
        return answer;
    }
}